### ISSUE TITLE
App Service Authentication CORS issue with BCP

### ISSUE THREAD

**Issue by SameerDoshi** [ChilliCream/hotchocolate#3854]
When App Service Auth is enabled on an app service running BCP BCP is unable to get schema related data due to a CORS issue.
Here's a sample of the error.
Access to manifest at 'https://login.windows.net/common/oauth2/authorize?response_type=code+id_token&redirect_uri=https%3A%2F%mysite.azurewebsites.net%2F.auth%2Flogin%2Faad%2Fcallback&client_id=4<clientID>-c94ee2e731de&scope=openid+profile+email&response_mode=form_post&nonce=a886b168118040c3bd912c7875e4f2e7_20210622134124&state=redir%3D%252Fapi%252Fgraphql%252Fplayground%252Fmanifest.json' (redirected from 'https://mysite.azurewebsites.net/api/graphql/playground/manifest.json') from origin 'https://mysite.azurewebsites.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Here's an article that talks about addressing this issue by changing the UI to support sending access credentials:
http://jsandersblog.azurewebsites.net/2020/04/23/azure-app-service-authentication-cors-error-easy-auth/

Cause: 
This behavior can occur if they are using fetch within their application. When using EasyAuth, a “Cookie” header is passed with the “AppServiceAuthSession” token. If the header is missing, an “authorize” request will be sent from provider. Sample below.

Fix: 
By default, fetch doesn’t automatically send cookies. To resolve the issue, the customer will need to add the “credentials” property with either “same-origin” or “include” provided.

**Comment by michaelstaib** [comment]
Did you have a look at the BCP settings?

They also can be preconfigured from the server.


The following example integrates with the identity server bff flow.

```
 endpoints.MapGraphQL()
    .WithOptions(new() { Tool = { Credentials = DefaultCredentials.Include } })
    .WithMetadata(new BffLocalApiEndpointAttribute(true));
```

**Comment by sergiu-enache** [comment]
this is correct, if you use cookie auth the headers are not added to /graphql/manifest json (which will try to trigger the auth flow)
![image](https://user-images.githubusercontent.com/77912457/130220581-a695e02b-ef73-4015-83bf-b70525c92021.png)

compared to the request for /graphql/bcp-config.json which does include the headers
![image](https://user-images.githubusercontent.com/77912457/130221126-97bee05f-4c46-462d-9822-79fecd4637f7.png)

As a workaround I can use static files middleware and manually add at path wwwroot/graphql/manifest.json
{
  "short_name": "Banana Cake Pop",
  "name": "Banana Cake Pop",
  "icons": [
    {
      "src": "icon.ico",
      "sizes": "64x64 32x32 24x24 16x16",
      "type": "image/x-icon"
    }
  ],
  "start_url": ".",
  "display": "standalone",
  "theme_color": "#000000",
  "background_color": "#ffffff"
}
